### DISCUSSION TITLE
[ARCHIVED] Exercise 3 - User Interface

### DISCUSSION THREAD

**Discussion by periode** [periode/augmenting-gallery#5] [Homework]
- Complete the interface Unity homework:
- - Create a button in Screen Space to make an object appear or disappear
- - Create a Text in World space that is related to the 3D object you've chosen
- - Create a button in World Space which changes the content of the Text.

**Comment by miafan716** [comment]
https://user-images.githubusercontent.com/55967980/154145070-af4756d4-4029-4498-a9b9-7273f5705d27.mp4

Here is what I have for this week. The start flag is the _button in Screen Space_ and by clicking on it _Kirby_, _Text in World Space_ and _Button in World Space_ would appear and the _button in Screen Space_  itself would disappear. The _Button in World Space_ can change the object showing on the screen. 

I wanted to make it a roll of 3 so that everytime I click the button would give me a new character. But I only figured out how to make the function run once because I can only drag one game object for the function.

**Comment by yylinj** [comment]
https://user-images.githubusercontent.com/98582179/154158996-0f7dd514-fc7a-4fe0-8662-b1e3b1d3e4f6.mov

I uploaded a table and a Nutella milkshake, but for some reason, they don't have any color.  The "Nutella Shake" button is the _button in Screen Space_, by clicking on it, the milkshake appears. _Text in World Space_ is first "this is Nutella Milkshake," then by clicking on the blue _button on the world space_, it changed to "but it is white." 

I don't know how to add the script, so I made the appear and disappear function in the "button --> on click --> game object SetActive" inspector section of the canvas for the button. So it is just a one-time thing, it cannot reappear 😭.

**Comment by aidanmassie** [comment]
https://user-images.githubusercontent.com/60477421/154165888-0e9f6134-7047-4049-a195-62c4922ca769.mov
![Screen Shot 2022-02-16 at 12 18 22 AM](https://user-images.githubusercontent.com/60477421/154166199-4a95f805-0b1c-4d96-b676-0f7d4921d797.png)


I decided to again use the Star Wars X-Wing as my 3D model for this exercise. I made the button in Screen Space the on/off toggle for the 3D object while making the button and text both in World Space. As seen in the screen recording, the Text Toggle Button and Text both exist in the 3D world, and the X-Wing toggle button only appears when the project is in play mode.

I used these tutorials to help me along the way:
https://www.youtube.com/watch?v=uBJvOa78t-g
https://www.youtube.com/watch?v=xpcyZdyO5P8&t=235s

**Comment by fractalessense** [comment]
https://user-images.githubusercontent.com/98880710/154171038-08449bd6-967b-4c3f-b1cf-d922d9f76b05.mov

I loaded in two 3D trees and added purple texture. Upon clicking the screen button 'chop down tree', one of the trees disappears and is revealed the text 'This is my purple tree', and an in-world button that says 'change text'. When one clicks on it, the text changes to read 'Our purple tree!'. Wholesome :)

**Comment by palomabrites** [comment]
https://user-images.githubusercontent.com/98867529/154184514-6a6d13d4-de96-48a0-800c-7d5fd57aef47.mp4

Bite button in screen space makes teeth model appear. Text in world space is changed by pressing info button.

**Comment by vb1424** [comment]
A tale about comparing :P 

https://user-images.githubusercontent.com/98861299/154350265-b7031069-aa6c-4efa-8c5c-fbb45b839f6c.mov
<img width="1038" alt="Screen Shot 2022-02-16 at 9 19 20 PM" src="https://user-images.githubusercontent.com/98861299/154350299-cd17e1a7-571c-45ce-ba80-060aecd7a8ee.png">

**Comment by vb1424** [comment]
https://user-images.githubusercontent.com/98861299/154350416-f639fb48-09ed-4d80-bb33-2ae91e1ce008.mov

**Comment by luyumenq** [comment]
https://user-images.githubusercontent.com/72938590/192304648-51e49aff-af24-4486-91c2-08a597a052ec.mov

**Comment by kenparks12** [comment]
https://user-images.githubusercontent.com/97991944/192537076-3f96df28-b558-4f30-b910-e1615927e39f.mov

<img width="1027" alt="Screen Shot 2022-09-27 at 15 09 38" src="https://user-images.githubusercontent.com/97991944/192537119-570637cd-8d1a-4173-941e-92a582f1fa43.png">
<img width="1028" alt="Screen Shot 2022-09-27 at 15 09 23" src="https://user-images.githubusercontent.com/97991944/192537125-f6dbbcf6-66ff-4646-bda0-60634a25fc64.png">
<img width="1027" alt="Screen Shot 2022-09-27 at 15 09 09" src="https://user-images.githubusercontent.com/97991944/192537131-d9e840a9-b8a8-4ea5-87b2-aca9b1111d29.png">

**Comment by IanPort** [comment]
https://user-images.githubusercontent.com/113472376/192656651-6488c516-1817-446e-b6cd-5dddd3bc8ca3.mov

**Comment by Alupie** [comment]
![IMG_7052](https://user-images.githubusercontent.com/113468588/192708895-bdbc5c4d-f4ea-4ec8-8842-f74c3163f669.jpg)
![IMG_7051](https://user-images.githubusercontent.com/113468588/192708900-f1529069-018f-43c0-8b5c-0cfd65352f65.jpg)
![IMG_7050](https://user-images.githubusercontent.com/113468588/192708903-6a48122c-1118-40b2-8b96-0115fb7aeba4.jpg)

I had a little trouble with the text so I made two different texts disappear. I wanted to get the text on the right to appear as the one on the left went away. I know this is some C# coding that I just didn't get the hang of yet.

**Comment by Dav1dyang** [comment]
<img width="1004" alt="Screen Shot 2022-09-28 at 9 03 37 AM" src="https://user-images.githubusercontent.com/55843005/192711335-9436ce4e-c4dd-4ce9-a5b4-3e259b9312b0.png">
The train moves into the frame. Arriving button will show the train arrived. Leaving button will show the train is delayed (as always).

Here's a link to the video: https://youtu.be/gsxWaTuMO5U

**Comment by kylewenchang** [comment]
https://user-images.githubusercontent.com/63214975/193920905-8551237b-38c1-4904-8ad8-6fd4e5a3dc65.mov

I made a guide to 3D shapes that shows some 3D shape properties when the mouse hovers over the shape. The spinning is done with a script, and the label updates with the button.

**Comment by aacarcrash** [comment]
<a href="https://youtu.be/75Iu7chONL4"> YouTube link to Project </a>

Project Title: Decolonising the Nandi Bull at the British Museum

Project Description: The project aims to tell the story of the Nandi Bull present in the British Museum. Where it originated from, to whom it passed to and where it is now. It also discusses the changes in its purposes, imagery and symbolisms as it changes locations. The project is set to Sachu Melung, a harsh Himalayan Anti Colonial Field Recording/soundscape made by Ruhail Qaisar, a sound artist from Leh, India.

**Reply by periode** [reply]
@AakSin 

It's a very atmospheric rendering of the life the Nandi bull, with a very clear, and yet smooth transition between three successive stages of the life of the object. You manage to turn a spatial exploration into a chronological exploration quite well.

That being said, this atmospheric rendering can sometimes be to the detriment of the didactic aspect of the piece. Not everyone in your audience might be as familiar as you with the colonial practices of the British Empire or with the concept of Exoticism/Orientalisms (unfortunately). Making some of this more explicit could help reach more people and drive your point home (for instance it could be something along the lines of "The Nandi bull was put to the mansion of the British governor, as it was considered a property of the Crown/the East India Company. Through this act, its status was relegated from being a religious object to a decorative object, contributing to certain set of fantasies of Western populations regarding distant lands.").

Finally, you might also want to mention the place of the Nandi bull and the position of the British Museum with regards to the conversation around restituting artworks.

Grade: 95 (A)

**Comment by Yupu-Chen** [comment]
https://user-images.githubusercontent.com/112383893/218863363-fc090cb9-bc8c-4da3-bdf6-e7c84ee2cebd.mov

This is part of my project for the first augmented exercise.

**Comment by Opal42** [comment]
**Enlightenment**
Encountering an artifact of unknown origin, this project seeks to poke fun at the difficult nature of trying to learn more about the background of an object, and the relative ease of simply "being in the moment", and making assumptions instead. At once it both challenges us to question the narratives offered in museums, while offering a brief glimpse into comparative techniques that can be used to help identify artifacts.

Google Drive link to both the Unity files and a video recording: https://drive.google.com/drive/folders/1KFIA3FDho_weSk3h1emvZfaUYGfX4Y3s?usp=sharing

**Reply by periode** [reply]
@Opal42 

Your approach to research and knowledge regarding this object was well done. You highlighted the fact that knowledge production sometimes appear as "magical", but is nonetheless a construct from particular individuals trying to figure things out. This lays bare the actual process of research and the actual role of museums in an academic sense, and it also highlights the difference between just looking at an object, and trying to look beyond it.

The presentation itself is also very appropriate, especially the diegetic tone that you give to the UI and the writing. We mentioned some of them in class, but a few ways forward would have been to put a bit more of a perceptual shift when going from "Enlightment" to "Ignorance" (for instance, animating the camera to signify a literal shift of perspective), provide an avenue for the visitor to suggest their own interpretation of the object (via the TextInput gameObject in unity) or by providing links to existing sources on the topic.

Grade: 95 (A)

**Comment by emersonmack** [comment]
![Screen Shot 2023-02-15 at 6 18 07 AM](https://user-images.githubusercontent.com/123585567/218940010-7626f9db-ee8d-4056-a297-59e42391342c.png)
![Screen Shot 2023-02-15 at 6 15 57 AM](https://user-images.githubusercontent.com/123585567/218940016-be91038e-0fb0-491e-9eb0-6a7c168cc978.png)
![Screen Shot 2023-02-15 at 6 16 42 AM](https://user-images.githubusercontent.com/123585567/218940019-1a3ef624-6b60-42c3-ac4d-a7e752d7e0d1.png)
![Screen Shot 2023-02-15 at 6 14 29 AM](https://user-images.githubusercontent.com/123585567/218940022-f0bdeef3-f8e9-4e82-b7c9-995c3731e5d4.png)
![Screen Shot 2023-02-15 at 6 15 03 AM](https://user-images.githubusercontent.com/123585567/218940025-529fca57-ca18-4462-b888-2ff57b94498b.png)
![Screen Shot 2023-02-15 at 6 10 48 AM](https://user-images.githubusercontent.com/123585567/218940028-87a62024-34fe-4dbc-8862-0a1cbd67e79d.png)

This interaction shows a gallery in 3D space where the user is able to click on buttons to change the presented information about the item. By selecting the information you'd like to obtain, this exhibition promotes an individualized approach to learning instead of throwing all information into one small text next to the object.
